sklearn/preprocessing/_encoders.py

```diff
@@ -618,6 +618,14 @@ class OrdinalEncoder(_BaseEncoder):
     dtype : number type, default np.float64
         Desired dtype of output.
 
+    handle_unknown : {'error', 'ignore'}, default='error'
+        Whether to raise an error or ignore if an unknown categorical feature
+        is present during transform (default is to raise). When this parameter
+        is set to 'ignore' and an unknown category is encountered during
+        transform, the resulting ordinal encoded value for this feature will be
+        -999. In the inverse transform, an unknown category will be denoted as
```

Comment:
I agree using -999 would be nice, but it would not really be "Ordinal". In other words, is an "unknown category" considered less than every other category?

In general, I think users want an "IntegerEncoder", which will map strings to ints with no assumption of order.

Reply:
Yes, I think an unknown category is different and, in a sense, considered less than other categories seen in the fit, because you have to treat them in a special way in a subsequent ML prediction (there were no parameters learned for an unknown category in the fit). Therefore, I think it should definitely not be an integer. Of course, -999 is an arbitrary choice and I have no preference here, but a negative number seems better than nan, as nan can be an original category (missing value) and would therefore be confusing.

Reply:
In the and one would wnat to have some explicit representation of "unknown". But for most algorithms we need a numerical type which do not have such an explicit representation (here integer). Then -999 actually has some advantages over some other choices:
- "0": very dangerous, as quite often this is interpreted as just the first "bin", also here if I see in correctly, so introducing this would be a breaking change
- "-1": better than 0, but it has the risk, that some might interpret this as belonging to the "0" bin, i.e. something like a overflow bin "lower than the lower boundary" which is not the same as "unknown"
- "-999": This is enough "far away" from any "accidental" collision, it is a very unnatural number, so we can quite safely abuse it to encode a type "unknown". Even if negative numbers are allowed, hitting -999 by chance is highly unlikely.

One more, but weak argument: In my working group at university years back this was a common practice and it "just worked" for years (disclaimer: @FelixWick was in the same group). 

One caveat might be: this might break some use cases, e.g if algorithms expect positive integers, so would it make sense to make this configurable? Even then, what would an alternative be in this case, "0"?

Reply:
0 is not an alternative, as it is the first category. And I think -999 is not a problem for algorithms expecting positive integers, because it is mandatory to handle this case of an unknown category specifically before the algorithm call (e.g. see it to a neutral value).